src/gui/qgsaddattrdialog.cpp

```diff
@@ -138,7 +138,7 @@ QgsField QgsAddAttrDialog::field() const
            ( QVariant::Type ) mTypeBox->currentData( Qt::UserRole ).toInt(),
            mTypeBox->currentData( Qt::UserRole + 1 ).toString(),
            mLength->value(),
-           mPrec->value(),
+           mPrec->isVisible() ? mPrec->value() : 0,
```

Comment:
This approach is a bit dangerous, because if the method is called after the dialog has been accepted then the widget is no longer visible in any case. Better to also disable it when you hide it, and use isEnabled here instead

Reply:
Fixed.